src/main/java/org/opensearch/knn/plugin/rest/RestClearCacheHandler.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.plugin.rest;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.Strings;
+import org.opensearch.index.Index;
+import org.opensearch.knn.common.exception.KNNInvalidIndicesException;
+import org.opensearch.knn.plugin.KNNPlugin;
+import org.opensearch.knn.plugin.transport.ClearCacheAction;
+import org.opensearch.knn.plugin.transport.ClearCacheRequest;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+
+import static org.opensearch.action.support.IndicesOptions.strictExpandOpen;
+import static org.opensearch.knn.common.KNNConstants.CLEAR_CACHE;
+import static org.opensearch.knn.index.KNNSettings.KNN_INDEX;
+
+/**
+ * RestHandler for k-NN Clear Cache API. API provides the ability for a user to evict those indices from Cache.
+ */
+public class RestClearCacheHandler extends BaseRestHandler {
+    private static final Logger logger = LogManager.getLogger(RestClearCacheHandler.class);
+
+    private static final String INDEX = "index";
+    public static String NAME = "knn_clear_cache_action";
+    private IndexNameExpressionResolver indexNameExpressionResolver;
+    private ClusterService clusterService;
+
+    /**
+     * Constructor
+     *
+     * @param clusterService  ClusterService
+     * @param indexNameExpressionResolver IndexNameExpressionResolver
+     */
+    public RestClearCacheHandler(ClusterService clusterService, IndexNameExpressionResolver indexNameExpressionResolver) {
```

Comment:
Should be possible to use lombok AllArgsConstructor, if you declare indexNameExpressionResolver and clusterService as final